### PR TITLE
Fixes some bad names and add new docs.

### PR DESCRIPTION
**Pull request by LambdAurora**
Fixed the badly named `GoToOwnerAndPurrGoal` which is doing something else entirely in reality.
Renamed sleeping with owner methods in `CatEntity` as an owner is not always present and is just putting the cat into a sleeping pose.
And added some javadocs too.

### CODE REVIEW

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/entity/passive/CatEntity.mapping [hunk 1]
```diff
@@ -31,11 +45,16 @@ CLASS net/minecraft/class_1451 net/minecraft/entity/passive/CatEntity
 		ARG 0 map
 	METHOD method_16084 updateHeadDownAnimation ()V
 	METHOD method_16085 updateAnimations ()V
-	METHOD method_16086 isSleepingWithOwner ()Z
+	METHOD method_16086 isInSleepingPose ()Z
+		COMMENT Returns whether this cat is in a sleeping pose.
+		COMMENT
+		COMMENT @return {@code true} if this cat is in a sleeping pose, otherwise {@code false}
```
Comment (liach) [inline mappings/net/minecraft/entity/passive/CatEntity.mapping]:
```suggestion
		COMMENT {@return whether this cat is in a sleeping pose}
```

**Review by haykam821 — CHANGES_REQUESTED** [review]
Other than the missing inline return tag previously noted, this pull request looks good.

File: mappings/net/minecraft/entity/passive/CatEntity.mapping (hunk 1 above)
Comment (haykam821) [reply mappings/net/minecraft/entity/passive/CatEntity.mapping]: Relates to #2454

**Review by YanisBft — APPROVED** [review]

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/entity/passive/CatEntity.mapping [hunk 2]
```diff
@@ -11,16 +11,30 @@ CLASS net/minecraft/class_1451 net/minecraft/entity/passive/CatEntity
...
 	FIELD field_30313 TABBY_TYPE I
+		COMMENT The tabby cat type, whose value is {@value}.
 	FIELD field_30314 BLACK_TYPE I
+		COMMENT The black cat type, whose value is {@value}.
 	FIELD field_30315 RED_TYPE I
+		COMMENT The red cat type, whose value is {@value}.
 	FIELD field_30316 SIAMESE_TYPE I
+		COMMENT The siamese cat type, whose value is {@value}.
 	FIELD field_30317 BRITISH_SHORTHAIR_TYPE I
+		COMMENT The british shorthair cat type, whose value is {@value}.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/passive/CatEntity.mapping]:
```suggestion
		COMMENT The British shorthair cat type, whose value is {@value}.
```

File: mappings/net/minecraft/entity/passive/CatEntity.mapping [hunk 3]
```diff
@@ -11,16 +11,30 @@ CLASS net/minecraft/class_1451 net/minecraft/entity/passive/CatEntity
...
 	FIELD field_30318 CALICO_TYPE I
+		COMMENT The calico cat type, whose value is {@value}.
 	FIELD field_30319 PERSIAN_TYPE I
+		COMMENT The persian cat type, whose value is {@value}.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/passive/CatEntity.mapping]:
```suggestion
		COMMENT The Persian cat type, whose value is {@value}.
```

File: mappings/net/minecraft/entity/passive/CatEntity.mapping [hunk 4]
```diff
@@ -11,16 +11,30 @@ CLASS net/minecraft/class_1451 net/minecraft/entity/passive/CatEntity
 	FIELD field_16291 prevSleepAnimation F
 	FIELD field_16292 HEAD_DOWN Lnet/minecraft/class_2940;
 	FIELD field_30313 TABBY_TYPE I
+		COMMENT The tabby cat type, whose value is {@value}.
 	FIELD field_30314 BLACK_TYPE I
+		COMMENT The black cat type, whose value is {@value}.
 	FIELD field_30315 RED_TYPE I
+		COMMENT The red cat type, whose value is {@value}.
 	FIELD field_30316 SIAMESE_TYPE I
+		COMMENT The siamese cat type, whose value is {@value}.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/passive/CatEntity.mapping]:
```suggestion
		COMMENT The Siamese cat type, whose value is {@value}.
```

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**